### ISSUE TITLE
Roman numeral page numbering in PDFs is not understood

### ISSUE THREAD

**Issue by willricemusic** [koreader/koreader#4245]
* KOReader version: v2015.11-1800-g0924b57_2018-07-29
* Device: Kobo Aura One

#### Issue
When viewing PDFs with roman numeral page numbering for the preface (e.g. page numbers run - i, ii, iii, iv, v, 1, 2, 3, 4 etc.) the roman numeral pages are:

1) Ignored by status bar > current page (instead numbering is shown only decimally, indexed to the first page of the document)

2) Not honoured when using the "Go to Page" dialog (e.g. I am reading a PDF textbook with 22 roman numeral preface pages before the numbering starts at 1. If I use "go to page" and enter 100, I in fact end up on page 77. I must remember to always add 23 to page numbers when navigating to account for the roman numeral preface).

#### Steps to reproduce
Attempt to navigate a PDF with roman numeral numbered pages at the beginning.

#### Expected behaviour
The status bar shows which page you are on in a PDF, whether it is a decimal page, or roman numeral page.

The "Go to Page" dialog accepts, and understands both decimal and roman numeral page numbers and navigates you to the correct page accordingly.

**Comment by Frenzie** [comment]
A similar issue exists with DjVu files where we plainly haven't implemented the relevant functionality. I forget for PDF and I don't have a device at hand right now, are they displayed correctly in the ToC?

**Comment by willricemusic** [comment]
> are they displayed correctly in the ToC?

No, they're indexed to the first page in the document (as in the status bar).

**Comment by Frenzie** [comment]
Upstream issue for PDF: https://bugs.ghostscript.com/show_bug.cgi?id=695351

Unfortunately there hasn't been any progress even though a patch exists.

**Comment by FSpark** [comment]
> Upstream issue for PDF: https://bugs.ghostscript.com/show_bug.cgi?id=695351
> 
> Unfortunately there hasn't been any progress even though a patch exists.

Just a reminder that there is already a new patch for version 1.18.0, but it still doesn't seem to be merged into the main branch either. Sad about the impact coming upstream.

**Comment by Frenzie** [comment]
Excellent! Now we just need to look real sweet at @benoit-pierre who seems to have the motivation to invest the required time. ;-D

**Comment by benoit-pierre** [comment]
There are builds with 1.22.0 [here](https://github.com/benoit-pierre/koreader/actions/runs/4839410226) if you want to test ;).

**Comment by Frenzie** [comment]
Dang, that looks pretty sexy. The git subtree/meson stuff, that is, didn't check the binaries. :-P

**Comment by Frenzie** [comment]
For quicker future reference:
```c
 /**
	Get page label for a given page.
*/
const char *fz_page_label(fz_context *ctx, fz_page *page, char *buf, int size);

/**
```
